src/components/adaptive-cards/AdaptiveCard/AdaptiveCard.jsx

```diff
@@ -85,6 +86,31 @@ export default function AdaptiveCard({
   const data = templateInstance.expand({
     $root: context,
   });
+
+  const [elementsProps, SetElementsProps] = useState({});
+
+  const setElementProps = useCallback((element) => {
+    SetElementsProps((prevElementProps) => (
+      {...prevElementProps, [element.id]: element}
+    ));
+  }, [SetElementsProps]);
+
+  const setPropValueForElement = (id, prop, value) => {
+    SetElementsProps((prevTargetElements) => {
+      const targetElem = prevTargetElements[id];
+
+      return {
+        ...prevTargetElements,
+        [id]: {...targetElem, [prop]: value},
+      };
+    });
+  };
+
+  const getValueOfProp = (id, prop) => ((id in elementsProps && prop in elementsProps[id])
+    ? elementsProps[id][prop]
+    : false
```

Comment:
this is a problem with a generic function, for other props the default value might not be `false`